lisa/sut_orchestrator/hyperv/source.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import Any, List, Type
+
+from lisa import schema
+from lisa.node import RemoteNode
+from lisa.tools.ls import Ls
+from lisa.tools.remote_copy import RemoteCopy
+from lisa.tools.unzip import Unzip
+from lisa.util import InitializableMixin, subclasses
+from lisa.util.logger import get_logger
+
+from .schema import LocalSourceSchema, SourceFileSchema, SourceSchema
+
+
+class Source(subclasses.BaseClassWithRunbookMixin, InitializableMixin):
+    def __init__(self, runbook: SourceSchema, **kwargs: Any) -> None:
+        super().__init__(runbook=runbook)
+        self._log = get_logger(
+            "source", self.__class__.__name__, parent=kwargs.get("parent_logger")
```

Comment:
parent could be moved to `__init__` like `def __init__(self, runbook: SourceSchema, parent_logger: Logger, **kwargs: Any) -> None:`

Reply:
This class is created by factory. So, we will need to add it in few other places too like `create_by_runbook()`, `create_with_runbook()` etc...

Reply:
The `create_by_runbook` has `**kwargs: Any`, so it should work.

Reply:
done.